packages/react/src/Grid/Grid.tsx

```diff
@@ -22,17 +22,76 @@ export type GridColumnNumbers = {
   wide: GridColumnNumber
 }
 
-interface GridComponent
-  extends ForwardRefExoticComponent<PropsWithChildren<HTMLAttributes<HTMLDivElement>> & RefAttributes<HTMLDivElement>> {
+type GridDensity = 'low' | 'high'
+
+type GridMarginSize = 'small' | 'medium' | 'large'
+
+type GridMarginVerticalProp = {
+  marginBottom?: never
+  marginTop?: never
+  /** The amount of vertical whitespace above and below the grid. */
+  marginVertical?: GridMarginSize
+}
+
+type GridMarginTopAndBottomProps = {
+  /** The amount of vertical whitespace below the grid. */
+  marginBottom?: GridMarginSize
+  /** The amount of vertical whitespace above the grid. */
+  marginTop?: GridMarginSize
+  marginVertical?: never
+}
+
+export type GridProps = {
+  /**
+   * The density of the grid: low (for websites) or high (for applications).
+   * Adjusts the pace with which columns get wider, and the start width as well.
+   * This is to be implemented more generally – it will be moved into a theme soon.
+   */
+  density?: GridDensity
+} & (GridMarginVerticalProp | GridMarginTopAndBottomProps) &
+  PropsWithChildren<HTMLAttributes<HTMLDivElement>>
+
+const marginClasses = (
+  marginBottom?: GridMarginSize,
+  marginTop?: GridMarginSize,
+  marginVertical?: GridMarginSize,
+): string[] => {
+  const classes = [] as string[]
+
+  if (marginVertical) {
+    return [`amsterdam-grid--margin-vertical--${marginVertical}`]
+  }
+
+  if (marginBottom) {
+    classes.push(`amsterdam-grid--margin-bottom--${marginBottom}`)
+  }
+
+  if (marginTop) {
+    classes.push(`amsterdam-grid--margin-top--${marginTop}`)
+  }
+
+  return classes
+}
+
+interface GridComponent extends ForwardRefExoticComponent<GridProps & RefAttributes<HTMLDivElement>> {
   Cell: typeof GridCell
 }
 
 export const Grid = forwardRef(
   (
-    { children, className, ...restProps }: PropsWithChildren<HTMLAttributes<HTMLDivElement>>,
+    { children, className, density = 'low', marginBottom, marginTop, marginVertical, ...restProps }: GridProps,
```

Comment:
If the default density is `low` shouldn't you rather use a boolean called `condesed` so you only have one modifier `--condesed` instead of having to define `density-low` and `density-high` everywhere?